KAFKA-17300: add document for new tiered storage feature in v3.9.0.

showuon opened the pull request:
1. Added document for disabling tiered storage at topic level
2. Added a notable change item in v3.9.0 for tiered storage quota.

The updated doc for "disabling tiered storage at topic level" looks like this:
![image](https://github.com/user-attachments/assets/2be2d6d0-a1af-476d-ada2-0e8122b2887f)


### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

showuon committed:
KAFKA-17300: add document for disabling tiered storage at topic level

showuon commented:
@kamalcph @satishd @clolov , document PR. Thanks.

kamalcph reviewed (approved):
LGTM

satishd commented on docs/ops.html:
```diff
@@ -4316,7 +4316,26 @@ <h4 class="anchor-heading"><a id="tiered_storage_config_ex" class="anchor-link">
 <pre><code class="language-bash">$ bin/kafka-console-consumer.sh --topic tieredTopic --from-beginning --max-messages 1 --bootstrap-server localhost:9092 --property print.offset=true</code></pre>
 
-<p>Please note, if you want to disable tiered storage at the cluster level, you should delete the tiered storage enabled topics explicitly.
+<p>If you want to disable tiered storage at the topic level to let the remote logs become read-only and no more local logs copied to the remote storage,
```
Please clarify that the disablement feature is available only with KRaft.

showuon replied:
Oh, good catch! Updated.

satishd reviewed (commented):
Thanks @showuon for the PR, left a comment.

showuon committed:
KAFKA-17300: mention only KRaft mode supports tiered storage disablement

showuon committed:
KAFKA-17300: add tiered storage quota and tiered storage GA in v3.9.0 notable change`

showuon commented:
@abhijeetk88 @satishd @kamalcph @clolov , in the document update PR, I added:

 1. Added document for disabling tiered storage at topic level
 2. Added a notable change item in v3.9.0 for tiered storage quota.
 3. Added a notable change item in v3.9.0 for making tiered storage feature as a general availability feature.

Please take a look when available. Thanks.

showuon committed:
KAFKA-17300: remove ga announcement

abhijeetk88 reviewed (approved):
LGTM

satishd reviewed (approved):
Thanks @showuon for addressing the review comments. LGTM.

kamalcph commented on docs/ops.html:
```diff
@@ -4316,7 +4316,28 @@ <h4 class="anchor-heading"><a id="tiered_storage_config_ex" class="anchor-link">
+<p>In KRaft mode, you can disable tiered storage at the topic level, to make the remote logs as read-only logs, or completely delete all remote logs.</p>
+
+<p>If you want to let the remote logs become read-only and no more local logs copied to the remote storage,
+  you can set <code>remote.storage.enable=true,remote.log.copy.disable=true</code> to the topic.</p>
```
```suggestion
  you can set <code>remote.storage.enable=true,remote.log.copy.disable=true,local.retention.ms=-2,local.retention.bytes=-2</code>
```

satishd commented:
A couple of unrelated test failures are observed in CI jobs. merging it to trunk.
